src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -13133,7 +14010,12 @@ void Assembler::cmovq(Condition cc, Register dst, Register src) {
 
 void Assembler::cmovq(Condition cc, Register dst, Address src) {
   InstructionMark im(this);
-  emit_int24(get_prefixq(src, dst), 0x0F, (0x40 | cc));
+  int prefix = get_prefixq(src, dst, true /* M0 */);
+  if (prefix_is_rex2(prefix)) {
+    emit_prefix_and_int8(prefix, (0x40 | cc));
+  } else {
+    emit_int24(prefix, 0x0F, (0x40 | cc));
+  }
```

Comment:
if  the newly added method get_prefixq(Address adr, bool isPage1) is modifed to include 0x0f as in comment there, then this block can be replaced by:
    emit_prefix_and_int8(prefix, (0x40 | cc));

Reply:
Very nice!  Done.